model/base_constraints.py

```diff
@@ -92,23 +94,22 @@ def add_maximum_one_shift_each_day(self, x):
         )
 
     def add_mapping_of_shift_to_demand(self, x, y):
-
         self.model.addConstrs(
             (
                 quicksum(x[e, t_marked, v] for t_marked, v in self.shifts_overlapping_t[t])
-                == quicksum(y[c, e, t] for c in self.competencies)
+                == quicksum(y[c, e, t] for c in self.competencies if y.get((c,e,t)))
                 for e in self.employees
-                for t in self.time_periods
+                for t in self.combined_time_periods
```

Comment:
I do not understand the changes here right away, could you please add an explanation?

Reply:
Combined means it is a combination of all the time periods in time_periods for all competencies. It is a set with time periods where there is demand without being linked to a competency

Reply:
So it is the original `time_periods`-set before adding competencies?

Reply:
I must have thought it was more difficult than that when fixing the code, but now I don't see any differences between them. That means the entire function for combining should be merged with the original get_time_periods function. Should be more efficient and cleaner than the current version.

Reply:
Is this an improvement you want to do now, or postpone to later?

Reply:
Postpone it. Not that critical and can be done in a later update.